helpers/search_processing_kendra_quickstart.sh

```diff
@@ -0,0 +1,174 @@
+fi
+
+DOCKER_BUILD_DIR=$(mktemp -d -p . opensearch-kendra-ranking-docker.XXXX)
+cd ${DOCKER_BUILD_DIR}
+
+echo "Running in $(pwd)"
+
+# Construct a Dockerfile that installs the search-processor plugin in the target image
+cat >Dockerfile <<EOF
+# Use base OpenSearch image
+FROM opensearchproject/opensearch:${OPENSEARCH_VERSION}
+
+# Install the plugin
+RUN /usr/share/opensearch/bin/opensearch-plugin install --batch https://github.com/opensearch-project/search-relevance/releases/download/${OPENSEARCH_VERSION}/search-processor.zip
+EOF
+
+# If credentials were found in the environment / profile, push them to the OpenSearch keystore
+# in the Docker image.
+if [ -n "${AWS_ACCESS_KEY_ID:-}" ] && [ -n "${AWS_SECRET_ACCESS_KEY:-}" ]; then
+  cat >>Dockerfile <<"EOF"
+
+# Push credentials to keystore
+RUN /usr/share/opensearch/bin/opensearch-keystore create
+RUN echo $AWS_ACCESS_KEY_ID | /usr/share/opensearch/bin/opensearch-keystore add kendra_intelligent_ranking.aws.access_key --stdin
+RUN echo $AWS_SECRET_ACCESS_KEY | /usr/share/opensearch/bin/opensearch-keystore add kendra_intelligent_ranking.aws.secret_key --stdin
+EOF
+  # If a session token was found in the environment / profile, push it to the OpenSearch keystore too.
+  if [ -n "${AWS_SESSION_TOKEN:-}" ]; then
+    echo "WARNING: Adding session token to OpenSearch keystore. These temporary credentials will"
+    echo "expire, and should only be used for short-lived testing."
+    echo 'RUN echo $AWS_SESSION_TOKEN | /usr/share/opensearch/bin/opensearch-keystore add kendra_intelligent_ranking.aws.session_token --stdin' >> Dockerfile
+  fi
+fi
+
+# Build and tag the Docker image with the plugin (and maybe credentials in the keystore)
+docker build --tag ${DOCKER_IMAGE_TAG} .
+
+# Create a docker-compose.yml file that will launch an OpenSearch node with the image we
+# just built and an OpenSearch Dashboards node that points to the OpenSearch node.
+cat >docker-compose.yml <<EOF
+version: '3'
+networks:
+  opensearch-net:
+services:
+  opensearch-node:
+    image: ${DOCKER_IMAGE_TAG}
+    container_name: opensearch-node
+    environment: 
+      - cluster.name=opensearch-cluster
+      - node.name=opensearch-node
+      - discovery.type=single-node
+      - kendra_intelligent_ranking.service.endpoint=${KENDRA_RANKING_ENDPOINT}
+      - kendra_intelligent_ranking.service.region=${AWS_REGION}
+      - kendra_intelligent_ranking.service.execution_plan_id=${EXECUTION_PLAN_ID}
+    ulimits:
+      memlock:
+        soft: -1
+        hard: -1
+      nofile:
+        soft: 65536
+        hard: 65536
+    ports:
+      - 9200:9200
+      - 9600:9600
+    networks:
+      - opensearch-net
+  opensearch-dashboard:
+    image: opensearchproject/opensearch-dashboards:${OPENSEARCH_VERSION}
+    container_name: opensearch-dashboards
+    ports:
+      - 5601:5601
+    environment:
+      OPENSEARCH_HOSTS: '["https://opensearch-node:9200"]'
+    networks:
+      - opensearch-net
+EOF
+docker-compose up
```

Comment:
For now, the docker-compose command is followed by a flurry of output for both containers (as they both flush to stdout on the current shell).  That's actually been kind of handy for me for debugging.

Unfortunately, it means that any helpful messages would get swept away.

I could switch this to `docker-compose up -d` to run in the background and offer those getting-started messages. Alternatively, instead of actually starting the containers, I could suggest that they go into the directory and run `docker-compose up` themselves, and suggest what they can do once the nodes are started.

I think the `-d` option makes more sense. I'll look into adding that in my next revision.

Reply:
I added some (hopefully) helpful text at the end.

It got a little long, so it outputs the text to a README file and then outputs that, with the last instruction being "If you ever want to read this text again, I've written it `<working_directory>/README`".

I covered your 4 suggestions, pointed to the docs for indexing, and explained how to add Kendra ranking to a query. I'm open to other ideas for things that might be useful.